### DISCUSSION TITLE
What was in the @cell.names section of Seurat_object V2?

### DISCUSSION THREAD

**Discussion by Jiayi-Zheng** [satijalab/seurat#6311] [Q&A]
Hello! I am new to cell clustering and am referencing some old codes for clustering analysis practice (in which they were using V2, but I'm using V4), I know the question seems quite easy but somehow I failed to find an answer to it...
Basically we merged two datasets of similar status and then created the seurat_object.

I encountered something as below in my reference code (Seurat v2): 
cell_name<-read.table(text=seurat_object@cell.names,sep="_",colClasses = "character")
age<-cell_name[,1]
names(age)<-seurat_object@cell.names
seurat_object<-AddMetaData(object = seurat_object, metadata = age, col.name = "age")

So I went to search for what is cell.names in V2 and got this: cell.names -> Names of all single cells (column names of the expression matrix)
Alongside this: V2 "seurat_object@cell.names" -> V3 "colnames(x = seurat_object)"

So, basically, cell.names is the barcodes for my cells, but that doesn't tell me what the "age" thing in my reference codes is. 

The barcodes are in format "1_AGAAGTATCCCGATCT-1", which there is an "_"... so the end product cell_name[,1] would be the "1". However, this would be exactly the same with orig.ident in meta.data, so I am confused why they have to repeat adding that column again into meta.data...Not sure if I interpreted something wrong and they've actually incorporated something else?

Thank you!
